### PR TITLE
Consolidate the tests in one class

### PR DESCRIPTION
**Pull request by skyluc**
Based on #12 

I will rebase on the other PRs when merged, but please review now.

### CODE REVIEW

**Comment by skonto** [comment]
I think there are two cases where we would like not to have one class for both cluster and client mode in terms of design:
a) we have created many tests so this class is getting huge 
b) we need to have specific testing for cluster vs client  modes. Then by context separation comes naturally.
Btw what does Int mean in MesosIntSpec name (integer tests)? Other than the above which may not be a concern now LGTM.

**Comment by skyluc** [comment]
@skonto, the problem is that code duplication is bad, and the current code is pretty bad because of the code duplication. Some code has been improved in one side, not on the other, ...

I'll rework it. We can split the test in different traits, split along group of test cases. Then the client and cluster test classes will be basically empty, extending the different traits defining the tests.

**Comment by skonto** [comment]
From a macro perspective the only difference is the mode boolean value passed at the runner along with the if statement. We could have one trait with common tests and one for cluster mode. So instead we could create a new dynamically with the appropriate mix-in (eg. predefined types).  Is this the idea you have?
From a micro perspective within each test we have similar functionality like spark ops eg rdd.sum etc which is the main test code and have different parameters or assertions for cluster vs client mode. Abstracting the test code seems ok, everything else may be too restrictive if that dictates a way to write test. Is there some implicit way from scalatest to deactivate certain tests when running one or the other mode while keeping consolidation of tests? If the list of tests grows big we can always split with some criteria.

**Comment by skyluc** [comment]
Yes, the macro level idea. So we don't have one class with all the tests, but we don't duplicate the test code for each model. Duplication inside the tests in not great, but not too bad at this point.

**Comment by skonto** [comment]
+1

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/mesosstate/StateDump.scala [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 import java.net.URL
 import java.io.InputStreamReader
 
-import org.typesafe.spark.mesos.tests.MesosIntTestHelper
+import com.typesafe.spark.test.mesos.MesosIntTestHelper
 
 object MesosState extends Enumeration {
   type MesosState = Value
-  val TASK_RUNNING, TASK_FINISHED = Value
+  val TASK_RUNNING, TASK_FINISHED, TASK_STAGING = Value
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/mesosstate/StateDump.scala]: Should we have all of them? `TASK_FAILED`, `TASK_ERROR`?

Comment (skyluc) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/mesosstate/StateDump.scala]: Yes we should. Did you find a spec/doc somewhere?

Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/mesosstate/StateDump.scala]: I think this is the [best we have](https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L1074)

**Comment by dragos** [comment]
LGTM

**Pull request merged**